### ISSUE TITLE
Bug - DataList - truncated content cannot be focused via keyboard

### ISSUE THREAD

**Issue by thatblindgeye** [patternfly/patternfly#7273]
**Describe the problem**
This is an issue with 
- [ ] Patternfly 5  
- [ ] Patternfly 6  
- [ ] other

The truncated content in examples (e.g. [text modifiers example](https://www.patternfly.org/components/data-list/html#text-modifiers)) cannot be navigated to via keyboard. While Core does not utilize tooltips like React examples, the truncated content should still be focusable and documentation should reflect that.

**How do you reproduce the problem?**
Go to the above example and try tabbing via keyboard, notice that the truncated content never receives focus.

**Expected behavior**
Truncated content receives focus via keyboard for a consumer to attach a tooltip to.

**Is this issue blocking you?**
List the workaround if there is one.

**Screenshots**
If applicable, add screenshots to help explain the issue.

**What is your environment?**

**What is your product and what release date are you targeting?**

**Any other information?**

**Comment by thatblindgeye** [comment]
Closing to consolidate into https://github.com/patternfly/patternfly/issues/7270
